### PR TITLE
Zach dev

### PR DESCRIPTION
**Pull request by zzwerling**
Ignore /src/components/Game.js

### CODE REVIEW

File: frontend/src/components/GameDetails/GameDetailsHeader.js [hunk 1]
```diff
@@ -87,8 +89,76 @@ const GameDetailsHeader = (props) => {
         if (props.data.involved_companies) {
             setGameDeveloper(props.data.involved_companies.filter((e) => e.developer == true)[0].company.name);
         }
+        getFavorites();
+
+
+        
+
+
```
Comment (elenirotsides) [inline frontend/src/components/GameDetails/GameDetailsHeader.js]:
```suggestion
```
Suggesting that you axe these extra spaces

**Review by patricksommer — CHANGES_REQUESTED** [review]
please mark this as a draft branch

**Review by zzwerling — COMMENTED** [review]
Merge conflicts dealt with, please approve @patricksommer

File: frontend/src/components/GameDetails/GameDetailsHeader.js [hunk 2]
```diff
@@ -86,8 +88,93 @@ const GameDetailsHeader = (props) => {
...
+            getFavorites();
+         })();
+         
+         
+
+       
     }, []);
 
+    useEffect(() =>{
+        if (curFavs.length > 0){
```
Comment (JordanH821) [inline frontend/src/components/GameDetails/GameDetailsHeader.js]: possibly looping over an error object

File: frontend/src/components/GameDetails/GameDetailsHeader.js [hunk 3]
```diff
@@ -86,8 +88,93 @@ const GameDetailsHeader = (props) => {
...
+            .then(
+                (result) => {
+                    setCurFavs(result);
+
+                   
+                },
+                (error) => {
+                    console.log(error);
+                    setCurFavs(error);
+                }
```
Comment (JordanH821) [inline frontend/src/components/GameDetails/GameDetailsHeader.js:224]: setting error object to the current favorites

File: frontend/src/components/GameDetails/GameDetailsHeader.js [hunk 4]
```diff
@@ -122,7 +209,7 @@ const GameDetailsHeader = (props) => {
             <div className={classes.leftContainer}>
                 <img className={classes.coverImage} src={props.data.cover.url}></img>
                 <div>
-                    <button class='btn btn-primary'>Add to favorites</button>
+                    <button onClick={async () => addRemoveFavorites(props.userData)}>{favText}</button>
```
Comment (JordanH821) [inline frontend/src/components/GameDetails/GameDetailsHeader.js]: I think this button styling should be included: `class='btn btn-primary'`

**Review by JordanH821 — CHANGES_REQUESTED** [review]

File: api/routes/users.js [hunk 5]
```diff
@@ -203,6 +203,42 @@ router.get('/picture', async function (req, res) {
     }
 });
 
+router.post('/modifyfavorites/:operation', async function(req, res){
+    let userId = req.body.userId;
+    let gameList = req.body.gameList;
+    let operation = req.params.operation;
+    let objId = new ObjectId(userId);
+    console.log(operation);
```
Comment (JordanH821) [inline api/routes/users.js]: remove this, only really log errors or weird edge cases

File: frontend/src/components/GameDetails/Game.js [hunk 6]
```diff
@@ -16,8 +16,13 @@ const Game = (props) => {
     const classes = styles();
     const [loading, setLoading] = useState(true);
     const [pageData, setPageData] = useState(undefined);
+    const [userData, setUserData] = useState(undefined);
     const [reloadReviews, setReloadReviews] = useState(false);
     let idToken;
+    let email = "zzwerlin@stevens.edu";
```
Comment (JordanH821) [inline frontend/src/components/GameDetails/Game.js]: remove

File: api/routes/users.js (hunk 5 above)
Comment (zzwerling) [reply api/routes/users.js]: removed the console.log

File: frontend/src/components/GameDetails/GameDetailsHeader.js (hunk 2 above)
Comment (zzwerling) [reply frontend/src/components/GameDetails/GameDetailsHeader.js]: ill set to null instead

File: frontend/src/components/GameDetails/GameDetailsHeader.js (hunk 3 above)
Comment (zzwerling) [reply frontend/src/components/GameDetails/GameDetailsHeader.js:224]: set to null instead

**Review by JordanH821 — APPROVED** [review]

**Review by JordanH821 — APPROVED** [review]

**Review by JordanH821 — APPROVED** [review]

**Review by patricksommer — APPROVED** [review]

**Pull request merged**